### ISSUE TITLE
No such file or directory:'/tmp/pyelastix/../result.0.mhd'

### ISSUE THREAD

**Issue by overfrequency** [almarklein/pyelastix#1]
Hi,

I get the following error:

```
In [26]:im3, field = pyelastix.register(im1, im2, params)
Calling Elastix to register images ...
---------------------------------------------------------------------------
RuntimeError                              Traceback (most recent call last)
<ipython-input-27-67b09ffdefda> in <module>()
----> 1 im3, field = pyelastix.register(im1, im2, params)

/usr/local/lib/python2.7/dist-packages/pyelastix.pyc in register(im1, im2, params, exact_params, verbose)
    507         except IOError as why:
    508             tmp = "An error occured during registration: " + str(why)
--> 509             raise RuntimeError(tmp)
    510
    511     # Find deformation field

RuntimeError: An error occured during registration: [Errno 2] No such file or directory: '/tmp/pyelastix/id_3862_140449827452240/result.0.mhd'
```

In that folder there is no result.0mhd file but there are im1.mhd,  im1.raw,  im2.mhd,  im2.raw, and  params.txt.

Have you ever see such an error or do you know how to solve it?

Thanks

Edit: 1. Environment is Ubuntu 14.04 and 16.04  and 2. Elastix works properly from command line :
elastix -f im1.png -m im2.png -out /tmp/pyelastix/xx -p /tmp/pyelastix/xx/params.txt

**Comment by lebedov** [comment]
I also observed the same error on MacOS 10.11.6 with Python 2.7.12 (installed via Anaconda), pyelastix 1.1, and the precompiled elastix 4.8 package. Running elastix from the command line also works properly.

**Comment by franfcunha** [comment]
Does anyone figured out how to solve this error?
Thanks

**Comment by lylos** [comment]
The problem is because the command is not formatted correctly to be passed to the command line.  Go to line 501 and change: command = [... to

``command = str(get_elastix_exes()[0])+' -m '+str(path_im1)+' -f '+str(path_im2)+' -out '+str(tempdir)+' -p '+str(path_params)``

This fixed it for me.

**Comment by eypros** [comment]
I had the same issue and your solution seems to make it work. It's really peculiar though that this wasn's fixed.
Anyway another line in deformation field needs modification also:
I used:

`command = '{} -def all -out {} -tp {}'.format(get_elastix_exes()[1], tempdir, path_trafo_params)`

instead of:
```
        command = [get_elastix_exes()[0],
                   '-m', path_im1, '-f', path_im2,
                   '-out', tempdir, '-p', path_params]

```

**Comment by avinashingit** [comment]
I tried changing the command but I'm getting this error "TabError: inconsistent use of tabs and spaces in indentation". I don't see any space misalignments but unsure why this error keeps coming.

**Comment by BenjaminAgthe** [comment]
Hey Guys!

I had the same problem running pyelastix 1.1, but changing the corresponding command = ... didn't solve my problem. 

First the tmp folder had no result.0.mhd but now it's deformationField.mhd

**[Errno 2] No such file or directory: '/tmp/pyelastix/id_27090_140626954330464/deformationField.mhd'**

Greetings and thx in advance

**Comment by zwep** [comment]
@avinashingit let me assure you that there is some inconsistent use of tabs and spaces... it can be very annoying to detect. Try to loading/reading the file with python, and look for "\t" and empty spaces in the string

**Comment by zwep** [comment]
> 
> 
> I had the same issue and your solution seems to make it work. It's really peculiar though that this wasn's fixed.
> Anyway another line in deformation field needs modification also:
> I used:
> 
> `command = '{} -def all -out {} -tp {}'.format(get_elastix_exes()[1], tempdir, path_trafo_params)`
> 
> instead of:
> 
> ```
>         command = [get_elastix_exes()[0],
>                    '-m', path_im1, '-f', path_im2,
>                    '-out', tempdir, '-p', path_params]
> ```

Err, I am confused by your use of get_elastix_exes()[1].. whereas the original is get_elastix_exes()[0].. is this intended?

**Comment by soupault** [comment]
I believe this is related to https://github.com/almarklein/pyelastix/pull/10 and has been fixed in my branch - https://github.com/almarklein/pyelastix/pull/20. I can no longer reproduce it with in my environment.

**Comment by soupault** [comment]
Closed in #20. Please, feel free to reopen if you face the issue again.
